src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,263 @@
+/* Copyright (C) 2015-2022, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "opBuilderHelperNetInfoAddress.hpp"
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include "syntax.hpp"
+
+#include "protocolHandler.hpp"
+#include <baseHelper.hpp>
+#include <kvdb/kvdbManager.hpp>
+#include <wdb/wdb.hpp>
+
+namespace
+{
+
+// TODO: remove pre release or leave it just for testing
+constexpr std::string_view STREAM_SOCK_PATH = "/tmp/testStream.socket";
+
+enum class Name
+{
+    ADDRESS,
+    NETMASK,
+    BROADCAST
+};
+
+constexpr std::string_view getPath(Name field)
+{
+    switch (field)
+    {
+        case Name::ADDRESS: return "/address";
+        case Name::NETMASK: return "/netmask";
+        case Name::BROADCAST: return "/broadcast";
+        default: return "";
+    }
+}
+
+// when isIPv6 == true the third field will be set to 0, otherwise 1 for IPv4
+// agent <agent_id> netaddr save <scan_ID>|<name>|isIPv6=false|add[i]|netm[i]|broad[i]
+// agent 0001 netaddr save 1234|name|0|add0|netm0|broad0
+bool sysNetAddresTableFill(base::Event event,
+                           const std::string& agentId,
+                           const std::string& scan_id,
+                           const std::string& name,
+                           const std::string& ipObjectPath,
+                           std::shared_ptr<wazuhdb::WazuhDB> wdb,
+                           bool isIPv6)
+{
+    // Cheking if AddresArray exists
+    std::optional<std::vector<json::Json>> address_ar;
+    try
+    {
+        address_ar = event->getArray(ipObjectPath + getPath(Name::ADDRESS).data());
+    }
+    catch (const std::runtime_error& e)
+    {
+        return false;
+    }
+
+    if (!address_ar.has_value())
+    {
+        return false;
+    }
+
+    std::string middleFieldName = isIPv6 ? "IPv6" : "IPv4";
+
+    std::optional<std::vector<json::Json>> netmaskI;
+    try
+    {
+        netmaskI = event->getArray(ipObjectPath + getPath(Name::NETMASK).data())
+                       .value_or(std::vector<json::Json>());
+    }
+    catch (const std::runtime_error& e)
+    {
+        netmaskI = std::nullopt;
+    }
+
+    std::optional<std::vector<json::Json>> broadcastI;
+    try
+    {
+        broadcastI = event->getArray(ipObjectPath + getPath(Name::BROADCAST).data())
+                         .value_or(std::vector<json::Json>());
+    }
+    catch (const std::runtime_error& e)
+    {
+        broadcastI = std::nullopt;
+    }
```

Comment:
std::nullopt vs empty array?,  is exception capture necessary?

Reply:
It could happen that
`ipObjectPath + getPath(Name::BROADCAST).data()`
Results in a wrong path throwing runtime_error